core/invoker/src/main/scala/whisk/core/invoker/InvokerReactive.scala

```diff
@@ -147,70 +156,74 @@ class InvokerReactive(
-                    case _                      => ActivationResponse.whiskError(Messages.actionRemovedWhileInvoking)
+    def processActivationMessage(bytes: Array[Byte]): Future[Unit] = {
+        Future(ActivationMessage.parse(new String(bytes, StandardCharsets.UTF_8)))
+            .flatMap(Future.fromTry(_))
+            .filter(_.action.version.isDefined)
+            .flatMap { msg =>
+                implicit val transid = msg.transid
+
+                val start = transid.started(this, LoggingMarkers.INVOKER_ACTIVATION)
+                val namespace = msg.action.path
+                val name = msg.action.name
+                val actionid = FullyQualifiedEntityName(namespace, name).toDocId.asDocInfo(msg.revision)
+                val subject = msg.user.subject
+
+                logging.info(this, s"${actionid.id} $subject ${msg.activationId}")
+
+                // caching is enabled since actions have revision id and an updated
+                // action will not hit in the cache due to change in the revision id;
+                // if the doc revision is missing, then bypass cache
+                if (actionid.rev == DocRevision.empty) {
+                    logging.warn(this, s"revision was not provided for ${actionid.id}")
                 }
-                val interval = Interval.zero
-                val causedBy = if (msg.causedBySequence) Parameters("causedBy", "sequence".toJson) else Parameters()
-                val activation = WhiskActivation(
-                    activationId = msg.activationId,
-                    namespace = msg.activationNamespace,
-                    subject = msg.user.subject,
-                    cause = msg.cause,
-                    name = msg.action.name,
-                    version = msg.action.version.getOrElse(SemVer()),
-                    start = interval.start,
-                    end = interval.end,
-                    duration = Some(interval.duration.toMillis),
-                    response = response,
-                    annotations = {
-                        Parameters("path", msg.action.toString.toJson) ++ causedBy
-                    })
-
-                activationFeed ! MessageFeed.Processed
-                ack(msg.transid, activation, msg.rootControllerIndex)
-                store(msg.transid, activation)
-        }
+
+                WhiskAction.get(entityStore, actionid.id, actionid.rev, fromCache = actionid.rev != DocRevision.empty).flatMap { action =>
+                    action.toExecutableWhiskAction match {
+                        case Some(executable) =>
+                            pool ! Run(executable, msg)
+                            Future.successful(())
+                        case None =>
+                            logging.error(this, s"non-executable action reached the invoker ${action.fullyQualifiedName(false)}")
+                            Future.failed(new IllegalStateException())
```

Comment:
We could do that for good measure. Note though that the message wouldn't be displayed anywhere (at least not today)